src/_pytest/fixtures.py

```diff
@@ -883,6 +885,8 @@ def call_fixture_func(
     fixturefunc: _FixtureFunc[FixtureValue], request: FixtureRequest, kwargs
 ) -> FixtureValue:
     if is_generator(fixturefunc):
+        # note: this also triggers on async generators, suppressing 'unawaited coroutine'
+        # warning.
         fixturefunc = cast(
             Callable[..., Generator[FixtureValue, None, None]], fixturefunc
```

Comment:
Perhaps we should rename this to `is_sync_generator` and update the implementation accordingly?  The type annotation and subsequent iteration code is simply incorrect for async generators...

Reply:
The one other place that `is_generator` is used is
https://github.com/pytest-dev/pytest/blob/76b8870f6cf04db8b405c766fc25021d3fc8602a/src/_pytest/python.py#L233-L241

and that check appears to be after checking if async, so should be fine to do a quick rename & implementation update.
Not that the implementation makes much sense anyway:
https://github.com/pytest-dev/pytest/blob/76b8870f6cf04db8b405c766fc25021d3fc8602a/src/_pytest/compat.py#L46-L48
(`iscoroutinefunction` does also check `getattr(func "_is_coroutine", False)` so it's not a complete no-op, but...)

That's maybe a separate PR as well though.

Reply:
...I think we should just delete this function and use `inspect.isgeneratorfunction` directly.  Happy to leave that for a future PR though.